joschahenningsen/TUM-Live
# Delisting unavailable pinned courses

### Motivation and Context
(https://github.com/TUM-Dev/gocast/issues/1351)

### Description
The code preloads the students when preloading the course, so that if a student is not enrolled at said course/doesn't have the view right the pinned lecture is not displayed anymore.

### Steps for Testing
Prerequisites:
- 1 Lecturer
- 1 Students
- 1 Lecture

1. Create a course
2. Enroll a student to said course
3. Pin the course as the student
4. Set the course as enrolled only and remove the student from database
5. Student should not see the course pinned anymore
